app/helpers/layout_helper.rb

```diff
@@ -347,6 +347,20 @@ def fullscreen_button(element = "$(this).prev()")
   private
 
   def authorized_associations(associations)
-    associations.included_modules.include?(Authorizable) ? associations.authorized : associations
+    if associations.included_modules.include?(Authorizable)
+      if associations.respond_to?(:klass)
+        associations.authorized(authorized_assocations_permission_name(associations.klass), associations.klass)
+      else
+        associations.authorized(authorized_assocations_permission_name(associations), associations)
+      end
+    else
+      associations
+    end
+  end
+
+  def authorized_assocations_permission_name(klass)
+    permission = "view_#{klass.to_s.underscore.pluralize}"
+    return nil unless Permission.where(:name => permission).present?
```

Comment:
it evolved from the 1.10 variant but makes more sense to write it as you suggested, I even added a comment even though this code won't be probably updated